include/mata/nfa/types.hh

```diff
@@ -48,6 +50,315 @@ struct Nfa; ///< A non-deterministic finite automaton.
 /// An epsilon symbol which is now defined as the maximal value of data type used for symbols.
 constexpr Symbol EPSILON = Limits::max_symbol;
 
+// defining indexes of logic operators used in an input vector for tseytin transformation
+#define TSEY_AND -1
+#define TSEY_OR -2
+#define TSEY_NOT -3
+
+// define charactes used for SAT and QBF solvers
+#define SOL_EOL std::string("0\n")
+#define SOL_NEG std::string("-")
+#define SOL_DELIM std::string(" ")
+#define SOL_HEADER std::string("p cnf ")
+#define SOL_FORALL std::string("a")
+#define SOL_EXISTS std::string("e")
+
+/**
+ * Base class for representing the input parameters for SAT and QBF reduction
+ */
+struct AutStats {
```

Comment:
The implementation of all of these structs should be somewhere else. These types should not be used by the user of the library, correct? Therefore, I think creating a header file and a source file for reduction (`reduction.cc` and `reduction.hh` and moving code of all reductions and all these structs inside is a reasonable approach).